app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java

```diff
@@ -340,23 +372,35 @@ private void zoomInLocation(LatLng center) {
      * The purpose of this application is to ask the user for their permission
      * of using their current location.
      */
-    private void getLocationPermission(){
+    private void getLocationPermission() {
         String[] permissions = {Manifest.permission.ACCESS_FINE_LOCATION,
                 Manifest.permission.ACCESS_COARSE_LOCATION};
-        if(requestPermission()){
-            myLocationPermissionsGranted = true;
-        }else{
-            ActivityCompat.requestPermissions(getActivity(),
-                    permissions,
-                    ClassConstants.LOCATION_PERMISSION_REQUEST_CODE);
+        if (ContextCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
what is happening in this if statement? I am kind of lost, putting some documentation would help!

Reply:
Even I, I am kind of confused since you seem to check if permission has not been granted, then you request it and verify again if it has been allowed otherwise you request it again? Breaking it down into sub-methods with clean names would be better in my opinion and would follow the clean code approach. This one seem to have a lot going on and is getting long

Reply:
I was trying to figure a solution for the opening on the current location however, nothing worked :"). i am cleaning it now.